### ISSUE TITLE
[llvm] build failure of mlir feature on Windows

### ISSUE THREAD

**Issue by rnett** [microsoft/vcpkg#23070]
**Host Environment**
- OS: Windows
- Compiler: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9 (this is the vpckg version, is that what you ment?)

**To Reproduce**
`./vcpkg install` with manifest:
```json
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "my-application",
  "version": "0.0.0",
  "dependencies": [
    {
      "name": "llvm",
      "default-features": false,
      "features": ["mlir"]
    }
  ]
}
```

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
```
Detecting compiler hash for triplet x86-windows...
The following packages will be built and installed:
    llvm[core,mlir,tools]:x86-windows -> 11.1.0
Could not locate cached archive: C:\Users\rnett\AppData\Local\vcpkg\archives\19\19ddad57a6a0745acfd321eb840f1f78aadf86e5.zip
Starting package 1/1: llvm:x86-windows
Building package llvm[core,mlir,tools]:x86-windows...
-- Note: llvm only supports static library linkage. Building static library.
-- Using cached C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/downloads/llvm-llvm-project-llvmorg-11.1.0.tar.gz
-- Cleaning sources at C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/buildtrees/llvm/src/org-11.1.0-a960fe6a14.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/downloads/llvm-llvm-project-llvmorg-11.1.0.tar.gz
-- Applying patch 0001-add-msvc-options.patch
-- Applying patch 0002-fix-install-paths.patch
-- Applying patch 0003-fix-openmp-debug.patch
-- Applying patch 0004-fix-dr-1734.patch
-- Applying patch 0005-fix-tools-path.patch
-- Applying patch 0006-workaround-msvc-bug.patch
-- Applying patch 0007-fix-compiler-rt-install-path.patch
-- Applying patch 0008-fix-libcxx-install.patch
-- Applying patch 0009-fix-tools-install-path.patch
-- Using source at C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/buildtrees/llvm/src/org-11.1.0-a960fe6a14.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: "C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe" --build . --config Debug --target install -- -v -j17
    Working Directory: C:/Users/rnett/Google Drive/My Stuff/kotlin-mlir-emitter/build/vcpkg/buildtrees/llvm/x86-windows-dbg
    See logs for more information:
      C:\Users\rnett\Google Drive\My Stuff\kotlin-mlir-emitter\build\vcpkg\buildtrees\llvm\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/llvm/portfile.cmake:225 (vcpkg_install_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package llvm:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: llvm:x86-windows
  Vcpkg version: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9

Additionally, attach any relevant sections from the log files above.
```
-Please attach any additional failure logs mentioned in the console output.
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/8054931/install-x86-windows-dbg-out.log)

**Additional context**
There's a bunch of path length warnings, so I enabled long paths, but this still occurs.  Tried installing non-manifest (`install llvm[core,mlir]`) with vcpkg in `C:\vcpkg`, still occurred.

**Comment by rnett** [comment]
Seems likely to be a LLVM 13 issue, I can install 12.0.1 (baseline commit `e6ed80cfca60da5ad016b28764818be7b7b0b77f`) just fine (although I'm only installing release builds now).

**Comment by rnett** [comment]
The error here was resolved by including `C++ ATL` in my Visual Studio installation.  Once that was fixed, I got the error from https://github.com/microsoft/vcpkg/issues/23077 (not exactly the same, because Windows, but still about `llvm::ThreadPool::isWorkerThread()`).  That was fixed by downgrading to 12.0.1 (builtin baseline `e6ed80cfca60da5ad016b28764818be7b7b0b77f`).

**Comment by PhoebeHui** [comment]
Closing in favor of https://github.com/microsoft/vcpkg/issues/23077.
